File: submission/akshay-sonar/src/builders/CustomerBuilder.ts
export interface Customer {
  id: string;
  email: string;
  name: string;
}

export class CustomerBuilder {
  private customer: Customer = {
    id: `customer_${Date.now()}`,
    email: "test@example.com",
    name: "Test Customer",
  };

  withId(id: string): this {
    this.customer.id = id;
    return this;
  }

  withEmail(email: string): this {
    this.customer.email = email;
    return this;
  }

  withName(name: string): this {
    this.customer.name = name;
    return this;
  }

  build(): Customer {
    return { ...this.customer };
  }
}

File: submission/karan/tests/framework/builders/customer_builder.py
from dataclasses import dataclass

from app.domain.models import Customer


@dataclass
class CustomerBuilder:
    customer_id: str = "cust_001"
    name: str = "Ada Lovelace"
    email: str = "ada@example.test"

    def build(self) -> Customer:
        return Customer(id=self.customer_id, name=self.name, email=self.email)
